core/protocols/aws-query-protocol/src/main/java/software/amazon/awssdk/protocols/query/interceptor/QueryParametersToBodyInterceptor.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.protocols.query.interceptor;
+
+import static java.util.Collections.singletonList;
+import static software.amazon.awssdk.utils.StringUtils.lowerCase;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.core.interceptor.Context;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.interceptor.ExecutionInterceptor;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.utils.CollectionUtils;
+import software.amazon.awssdk.utils.http.SdkHttpUtils;
+
+/**
+ * Modifies an HTTP request by moving query parameters to the body under the following conditions:
+ * - It is a POST request
+ * - There is no content stream provider
+ * - There are query parameters to transfer
+ *
+ * Observe that this interceptor should only be applied to the Query Protocol.
+ */
+@SdkInternalApi
+public final class QueryParametersToBodyInterceptor implements ExecutionInterceptor {
+
+    @Override
+    public SdkHttpRequest modifyHttpRequest(Context.ModifyHttpRequest context,
+                                                      ExecutionAttributes executionAttributes) {
+
+        SdkHttpRequest httpRequest = context.httpRequest();
+
+        if (!(httpRequest instanceof SdkHttpFullRequest)) {
+            return httpRequest;
+        }
```

Comment:
What are the situations under which this legitimately happens?

Reply:
I don't know :-). I put it in more defensively, you could say.